packages/docs/src/routes/docs/cookbook/debouncer/index.mdx

```diff
@@ -31,12 +31,17 @@ The Qwik framework provides unique capabilities for managing state and effects i
 
 <CodeFile src="/src/routes/demo/cookbook/mediaController/index.tsx">
 ```tsx
-export const useDebouncer = (fn: QRL<(args: any) => void>, delay: number) => {
+export const useDebouncer = <A extends readonly unknown[], R>(
```

Comment:
I've never used read-only on an args array, what does it do?

Reply:
Using it in a function definition is typically good, because it allows callers to pass readonly values:

```
function mutableArg(arr: unknown[]): void;
function readonlyArg(arr: readonly unknown[]): void;

const mutableArr: unknown[] = [];
const readonlyArr: readonly unknown[] = [];

mutableArg(mutableArr); // ✅
mutableArg(readonlyArr); // ❌

readonlyArg(mutableArr); // ✅
readonlyArg(readonlyArr); // ✅
```

In the case of `arguments` or `...args`, I don't think this impacts the caller at all. It just makes the arguments array read-only.